src/redecomp/RedecompMesh.cpp

```diff
@@ -54,20 +54,26 @@ RedecompMesh::RedecompMesh( const mfem::ParMesh& parent, double ghost_length, Pa
       SLIC_ERROR_ROOT( "Only 2D and 3D meshes are supported." );
   }
 
-  // preclude degenerate case where num elements/2 < num ranks
-  // factor of 2 on num elements are due to elements being paired for contact
-  auto n_parts = std::min( parent.GetNRanks(), ( static_cast<int>( parent.GetGlobalNE() ) + 1 ) / 2 );
-  p2r_elems_ = BuildP2RElementList( *partitioner, n_parts, ghost_length );
+  auto parent_n_els = static_cast<int>( parent.GetGlobalNE() );
+  if ( n_ranks == 0 ) {
+    // preclude degenerate case where num elements/2 < num ranks
+    // factor of 2 on num elements are due to elements being paired for contact
+    // additional factor of 10 targets about 10 pairs/rank
```

Comment:
Mention for clarification that this logic assumes conforming mesh, but that this is used as a lower limit. A nonconforming mesh would result in more than a minimum of 10 pairs per rank (or something like that).